crates/starknet-server/src/api/http/endpoints/dump_load.rs

```diff
@@ -1,19 +1,38 @@
 use axum::{Extension, Json};
 
 use crate::api::http::error::HttpApiError;
-use crate::api::http::models::Path;
+use crate::api::http::models::{DumpLoadResponse, Path};
 use crate::api::http::{HttpApiHandler, HttpApiResult};
 
 pub(crate) async fn dump(
-    Json(_path): Json<Path>,
-    Extension(_state): Extension<HttpApiHandler>,
-) -> HttpApiResult<()> {
-    Err(HttpApiError::GeneralError)
+    Json(path): Json<Path>,
+    Extension(state): Extension<HttpApiHandler>,
+) -> HttpApiResult<Json<DumpLoadResponse>> {
+    if path.path.is_empty() {
+        return Err(HttpApiError::PathNotFound);
```

Comment:
In Devnet-py we allowed path to be empty if it was specified with `--dump-path`:
https://github.com/0xSpaceShard/starknet-devnet/blob/ef036e26fac86ff197e8fecd0550f62776d511d2/starknet_devnet/blueprints/base.py#L91

Did you consult the Python implementation (also for tests)?

EDIT: I am now coming back to this comment confused after seeing the defaulting done with:
```rust
if custom_path.is_some() { &custom_path } else { &self.config.dump_path }
```

Reply:
I was briefly checking Python implementation at the beginning of that task but to be honest, this re-execution approach is a completely different way of doing this dump and load.

Fixed in 4d89670c7d9c891765ce73533e82e83eb0908c54